### PR TITLE
feat: Add health check for canvas integrated channels

### PR DESCRIPTION
**Pull request by marlonkeating**
[Jira Ticket](https://2u-internal.atlassian.net/browse/ENT-5607)

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [x] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [x] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

**Comment by johnnagro** [comment]
I really like the idea of wiring this up to an http endpoint. I wonder if we should standardize around the more generic `/config` URLs from here https://github.com/openedx/edx-enterprise/blob/master/integrated_channels/api/v1/urls.py#L18

maybe '/config/<uuid>/health` or something?

I think we should discuss standardizing on '/config' for more things?

File: integrated_channels/api/v1/urls.py [hunk 1]
```diff
@@ -2,9 +2,9 @@
 URL definitions for v1 Integrated Channel API endpoints.
 """
 
-from django.urls import include, path
+from django.urls import include, path, re_path
```
Comment (alex-sheehan-edx) [inline integrated_channels/api/v1/urls.py:5]: I don't think re_path is needed anymore

File: integrated_channels/api/v1/views.py [hunk 2]
```diff
@@ -78,3 +83,45 @@ def list(self, request, *arg, **kwargs):
         response = blackboard_serializer.data + canvas_serializer.data + cornerstone_serializer.data + \
             degreed_serializer.data + degreed2_serializer.data + moodle_serializer.data + sap_serializer.data
         return Response(response)
+
+
+class IntegratedChannelHealthCheckView(generics.ListAPIView):
+    """
+        **Use Cases**
```
Comment (alex-sheehan-edx) [inline integrated_channels/api/v1/views.py:90]: great docstring 😄

File: integrated_channels/api/v1/views.py [hunk 3]
```diff
@@ -78,3 +83,45 @@ def list(self, request, *arg, **kwargs):
...
+
+    @method_decorator(require_at_least_one_query_parameter('channel_code'))
+    @method_decorator(require_at_least_one_query_parameter('uuid'))
+    def get(self, request, *args, **kwargs):
+        channel_code = self.request.query_params.get('channel_code')
+        uuid = self.request.query_params.get('uuid')
+
+        # Get enterprise configuration
+        config_type = EnterpriseCustomerPluginConfiguration.get_class_by_channel_code(channel_code)
+        config = config_type.objects.get(uuid=uuid)
```
Comment (alex-sheehan-edx) [inline integrated_channels/api/v1/views.py:120]: some error handling around channel and config do not exist would be nice here

File: integrated_channels/api/v1/views.py [hunk 4]
```diff
@@ -78,3 +83,45 @@ def list(self, request, *arg, **kwargs):
...
+        client_type = get_enterprise_client_by_channel_code(channel_code)
+        client = client_type(config)
+
+        health_status = client.health_check()
+        is_healthy = health_status == IntegratedChannelHealthStatus.HEALTHY
+        payload = {'is_healthy': is_healthy, 'health_status': health_status.value}
+        return Response(payload, status=HTTP_200_OK)
```
Comment (alex-sheehan-edx) [inline integrated_channels/api/v1/views.py:127]: I think the linter will complain about no new lines?

File: integrated_channels/api/v1/views.py (hunk 3 above)
Comment (alex-sheehan-edx) [reply integrated_channels/api/v1/views.py:120]: if we're going to release this as is then also probably a list of supported channels with the validation?

**Review by alex-sheehan-edx — COMMENTED** [review]
A couple comments and we should probably have some tests to along with the new views. But otherwise I think your approach is great and this is looking good.

File: integrated_channels/api/v1/urls.py (hunk 1 above)
Comment (marlonkeating) [reply integrated_channels/api/v1/urls.py:5]: Good catch, thanks!

**Review by alex-sheehan-edx — APPROVED** [review]
Superb! ✅

**Pull request merged**